src/datatable/tpl/layout.tpl

```diff
@@ -83,25 +83,25 @@
                 </tr>
             </thead>
             <tbody>
-                {{#each dataset.data}}
+                {{#each dataset.data as |row|}}
                     <tr data-item-identifier="{{id}}">
-                        {{#if ../options.selectable}}
+                        {{#if @root.options.selectable}}
                         <td class="checkboxes"><input type="checkbox" name="cb[{{id}}]" value="1" /></td>
                         {{/if}}
 {{! IMPORTANT:START IF YOU'RE GOING TO MAKE CHANGES TO THIS SECTION,
     PLEASE UPDATE shallowUpdate METHOD ACCORDINGLY}}
-                        {{#each ../options.model}}
+                        {{#each @root.options.model}}
                             {{#if type}}
                             <td class="actions {{id}}">
-                                {{#each ../actions}}
+                                {{#each actions as |action|}}
                                     {{#if id}}
-                                        {{#with ../../../../this}}
-                                            {{#unless ../hidden}}
-                                <button class="btn-info small {{../../id}}"
-                                    {{#if ../../title}} title="{{../../../title}}"{{/if}}
-                                    {{#if ../../disabled}} disabled="disabled"{{/if}}>
-                                    {{#if ../../icon}}<span class="icon-{{../../../icon}}"></span>{{/if}}
-                                    {{../../label}}
+                                        {{#with row}}
```

Comment:
This `#with` is needed, because `action.hidden` is expected to be a function with reference to row as `this`:
```
actions: [{
  id: 'action-1', 
  hidden: function() {
    this.userFirstName === 'nobody'
  }
}]
```